buildSrc/src/main/java/org/opensearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -41,11 +41,10 @@
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
 import org.gradle.api.model.ObjectFactory;
-import org.gradle.api.provider.Provider;
 import org.gradle.api.provider.ProviderFactory;
 import org.gradle.internal.jvm.Jvm;
-import org.gradle.jvm.toolchain.JavaInstallation;
-import org.gradle.jvm.toolchain.JavaInstallationRegistry;
+import org.gradle.internal.jvm.inspection.JvmInstallationMetadata;
```

Comment:
Why does this use a package from `org.gradle.internal`? It seems this is not intended for usage and not documented in the Javadocs.

I do see that this made it into `main` and this is just the backport.

Reply:
@dlvenable the Opensearch build manipulates JVM installations on several occasions, Gradle 7 changes the underlying implementation significantly, so the 2 options we had: a) find the way to keep manipulating JVM installations (sadly, using internal classes) b) scrape and rewrite large chunks of the build (using toolchains or alike). The first option looked like reasonable tradeoff from efforts perspective, but I would be happy to see better solution.

Reply:
Thanks @reta for the explanation. That does sound like a reasonable tradeoff. I do think that a better long-term solution would be to use the new toolchains feature. It would take some upfront work, but should also allow easier updating of Gradle versions since it will not be tied to internal implementations.

Reply:
:+1: no objections!